clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java

```diff
@@ -131,67 +106,39 @@ public ClusteringServiceConfiguration(final URI clusterUri, final TimeoutDuratio
    * @param autoCreate {@code true} if server components should be auto created
    * @param serverConfig  the server side entity configuration required
    *
-   * @throws NullPointerException if {@code clusterUri} or {@code serverConfig} is {@code null}
+   * @throws NullPointerException if {@code clusterUri} is {@code null}
    * @throws IllegalArgumentException if {@code clusterUri} is not URI valid for cluster operations
    */
-  public ClusteringServiceConfiguration(final URI clusterUri, boolean autoCreate, ServerSideConfiguration serverConfig) {
-    validateClusterUri(clusterUri);
-    if (serverConfig == null) {
-      throw new NullPointerException("Server configuration cannot be null");
-    }
-    this.clusterUri = clusterUri;
-    this.autoCreate = autoCreate;
-    this.serverConfiguration = serverConfig;
-    this.readOperationTimeout = DEFAULT_READ_OPERATION_TIMEOUT;
+  public ClusteringServiceConfiguration(URI clusterUri, boolean autoCreate, ServerSideConfiguration serverConfig) {
+    this(clusterUri, Timeouts.DEFAULT, autoCreate, serverConfig);
   }
 
   /**
    * Creates a {@code ClusteringServiceConfiguration} from the properties provided.
    *
    * @param clusterUri the non-{@code null} URI identifying the cluster server
-   * @param readOperationTimeout the {@code TimeoutDuration} specifying the time limit for clustered cache
-   *                            read operations; if {@code null}, the default value is used
+   * @param timeouts the {@link Timeouts} specifying the time limit for clustered cache operations
    * @param autoCreate {@code true} if server components should be auto created
    * @param serverConfig  the server side entity configuration required
    *
-   * @throws NullPointerException if {@code clusterUri} or {@code serverConfig} is {@code null}
+   * @throws NullPointerException if {@code clusterUri} is {@code null}
    * @throws IllegalArgumentException if {@code clusterUri} is not URI valid for cluster operations
    */
-  public ClusteringServiceConfiguration(final URI clusterUri, final TimeoutDuration readOperationTimeout, boolean autoCreate, ServerSideConfiguration serverConfig) {
-    validateClusterUri(clusterUri);
-    validateReadOperationTimeout(readOperationTimeout);
-    if (serverConfig == null) {
-      throw new NullPointerException("Server configuration cannot be null");
-    }
-    this.clusterUri = clusterUri;
+  public ClusteringServiceConfiguration(URI clusterUri, Timeouts timeouts, boolean autoCreate, ServerSideConfiguration serverConfig) {
+    this.clusterUri = Objects.requireNonNull(clusterUri, "Cluster URI cannot be null");
     this.autoCreate = autoCreate;
     this.serverConfiguration = serverConfig;
```

Comment:
Haven't we ended up dropping the null-checking on serverConfig here?

Reply:
Why? It looks like good defensive programming.

Reply:
Yes. Because null is a valid serverConfig value